### ISSUE TITLE
Getters taking int should be renamed

### ISSUE THREAD

**Issue by jlconlin** [njoy/ENDFtk#59]
We have a number of "getters" that don't feel a little awkward. For example:
```c++
decltype( auto ) material = tape.materialNumber( 125 );
```
This feels like the type of `material` would be an `int` (a material *number*). Similarly:
```c++
decltype( auto ) material = tape.materialNumber( 125 );
decltype( auto ) file = material.fileNumber ( 3 );
decltype( auto ) section = file.sectionNumber( 102 );
```
Because of the name of the method, it feels like these should return an `int`.

I propose that we should rename these functions and remove the `Number` at the end of each function name, such as:
```c++
decltype( auto ) material = tape.material( 125 );
decltype( auto ) file = material.file ( 3 );
decltype( auto ) section = file.section( 102 );
```
This makes it feel like the return type will be a `Material`, `File`, or `Section` respectively. 

This is a relatively big API change. However, it is just a search/replace in the current repository. There is only one other component that currently uses ENDFtk (to my knowledge) so we should decide what is the best way to do this and do it right before it becomes too difficult to change it.

**Comment by jlconlin** [comment]
Part of this discussion should be what should we do with the associated methods:
```c++
decltype( auto ) material = tape.MAT( 125 );
decltype( auto ) file = material.MF ( 3 );
decltype( auto ) section = file.MT( 102 );
```
What should these return?

**Comment by whaeck** [comment]
These equivalent functions on file::Type have been changed some time ago to be more "logical" already. sectionNumber(...) is already section(...) which returns a section::Type, with MT(...) being an alias for this function.

These changes were made when the MF6 coding was added in (we refactored the section::Type and file::Type heavily at that time).

Using sectionNumber() or MT() (without any argument) should return the actual section number, and should be methods on the object representing a section with a given MT number. Using section(...) and MT(...) (with an int or literal constant argument) should return a section object (or throw an exception if the MT number is not present in the file), and should be methods on the object representing a file containing one or more sections.

**Comment by whaeck** [comment]
This has been addressed in #92 

There now is a distinction between file(...) and fileNumber(), the first one returns the file, the second one the MF number. Same was done for material and section (in both the ENDF tree and the ful components).

**Comment by whaeck** [comment]
Fixed in #92
